src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -334,6 +432,227 @@ public int GetCount(bool onlyIfCheap)
+                        state = 2;
+                        goto case 2;
+                    case 2:
+                        if (_enumerator.MoveNext())
+                        {
+                            current = _selector(_enumerator.Current);
+                            return true;
+                        }
+                        Dispose();
+                        break;
+                }
+                return false;
+            }
+
+            public override void Dispose()
+            {
+                if (_enumerator != null)
+                {
+                    _enumerator.Dispose();
+                    _enumerator = null;
+                }
+                base.Dispose();
+            }
+
+            public override IEnumerable<TResult2> Select<TResult2>(Func<TResult, TResult2> selector)
+            {
+                return new SelectIPartitionIterator<TSource, TResult2>(_source, CombineSelectors(_selector, selector));
+            }
+
+            public IPartition<TResult> Skip(int count)
+            {
+                return count == 0 ? (IPartition<TResult>)Clone() : new SelectIPartitionIterator<TSource, TResult>(_source.Skip(count), _selector);
+            }
+
+            public IPartition<TResult> Take(int count)
+            {
+                return new SelectIPartitionIterator<TSource, TResult>(_source.Take(count), _selector);
+            }
+
+            public bool TryGetElementAt(int index, out TResult element)
+            {
+                TSource input;
+                if (_source.TryGetElementAt(index, out input))
+                {
+                    element = _selector(input);
+                    return true;
+                }
+
+                element = default(TResult);
+                return false;
+            }
+
+            public bool TryGetFirst(out TResult first)
+            {
+                TSource input;
+                if (_source.TryGetFirst(out input))
+                {
+                    first = _selector(input);
+                    return true;
+                }
+
+                first = default(TResult);
+                return false;
+            }
+
+            public bool TryGetLast(out TResult last)
+            {
+                TSource input;
+                if (_source.TryGetLast(out input))
+                {
+                    last = _selector(input);
+                    return true;
+                }
+
+                last = default(TResult);
+                return false;
+            }
+
+            public TResult[] ToArray()
+            {
+                if (typeof(TSource) == typeof(TResult))
+                {
+                    TSource[] source = _source.ToArray();
+                    TResult[] result = (TResult[])(Array)source;
+
+                    for (int i = 0; i != source.Length; ++i)
+                        result[i] = _selector(source[i]);
+
+                    return result;
+                }
```

Comment:
Hmm. That's annoying. I'd put it in only to cover this types worse case (can't get a constant-time count) and been surprised by how good it turned out to be in practice to the point where I was going to see if it would improve other cases.

But yes, this one definitely has to go.